pubsub/server.go

```diff
@@ -0,0 +1,75 @@
+package pubsub
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"errors"
+	"sync"
+
+	"github.com/NethermindEth/juno/utils"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/sourcegraph/conc"
+)
+
+var ErrNotFound = errors.New("subscription not found")
+
+type Server struct {
+	log utils.SimpleLogger
+	wg  conc.WaitGroup
+
+	mu            sync.Mutex // protects subscriptions
+	subscriptions map[uint64]event.Subscription
+}
+
+func New(log utils.SimpleLogger) *Server {
+	return &Server{
+		subscriptions: make(map[uint64]event.Subscription),
+		log:           log,
+	}
+}
+
+func getRandomID() uint64 {
+	var n uint64
+	for err := binary.Read(rand.Reader, binary.LittleEndian, &n); err != nil; {
+	}
+	return n
+}
+
+func (s *Server) Subscribe(ctx context.Context, sub event.Subscription) uint64 {
+	id := getRandomID()
```

Comment:
why random? could we get away with an incrementing counter?

Reply:
I'm almost positive I read about some security issue related to sequential ids. I can't find it right now. Will leave this open for now.

Reply:
I guess the only issue is leaking information about how many subscriptions have been served. Would you like it to be incrementing? I don't mind either way.